packages/jellyfish-api-core/__tests__/category/icxorderbook/listOrders.test.ts

```diff
@@ -0,0 +1,328 @@
+
+    // List the ICX offers for orderTx = createOrderTxId and check
+    orders = await client.icxorderbook.listOrders({ orderTx: createOrderTxId })
+    expect(Object.keys(orders).length).toBe(3) // extra entry for the warning text returned by the RPC atm.
+    await checkDFIBuyOfferDetails(container, offer, makeOfferTxId, orders as Record<string, ICXOfferInfo>)
+    await checkDFIBuyOfferDetails(container, offer2, makeOffer2TxId, orders as Record<string, ICXOfferInfo>)
+
+    // now close makeOffer2TxId
+    await client.icxorderbook.closeOffer(makeOffer2TxId)
+    await container.generate(1)
+
+    // List the ICX offers for orderTx = createOrderTxId and check
+    orders = await client.icxorderbook.listOrders({ orderTx: createOrderTxId })
+    expect(Object.keys(orders).length).toBe(2) // extra entry for the warning text returned by the RPC atm.
+    await checkDFIBuyOfferDetails(container, offer, makeOfferTxId, orders as Record<string, ICXOfferInfo>)
+
+    // now list the ICX offers for orderTx = createOrderTxId with ICXListOrderOptions.closed=true and check
+    orders = await client.icxorderbook.listOrders({ orderTx: createOrderTxId, closed: true })
+    expect(Object.keys(orders).length).toBe(2) // extra entry for the warning text returned by the RPC atm.
+    const offerInfoRetrived = orders as Record<string, ICXOfferInfo>
+    expect(offerInfoRetrived[makeOffer2TxId].orderTx).toStrictEqual(offer2.orderTx)
+    expect(offerInfoRetrived[makeOffer2TxId].status).toStrictEqual(ICXOrderStatus.CLOSED)
+    expect(offerInfoRetrived[makeOffer2TxId].ownerAddress).toStrictEqual(offer2.ownerAddress)
+  })
+
+  // NOTE(surangap): try to write a parameterized test for all invlaid inputs of ICXListOrderOptions. Ideally client.icxorderbook.listOrders()
+  // should return empty set for all such cases. but at the moment C++ side implemenation does not go with that.
+
+  it('Should return an empty set when ICXListOrderOptions.orderTx is invalid', async () => {
+    // create an order - maker
+    const order: ICXOrder = {
+      tokenFrom: idDFI,
+      chainTo: 'BTC',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    let result: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = result.txid
+    await container.generate(1)
+
+    // create offer to order createOrderTxId
+    const offer: ICXOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.1), // 10 DFI = 0.1 BTC
+      ownerAddress: accountBTC
+    }
+    result = await client.icxorderbook.makeOffer(offer, [])
+    await container.generate(1)
+
+    // list offers for Tx Id "123" and check
+    const orders: Record<string, ICXOrderInfo| ICXOfferInfo> = await client.icxorderbook.listOrders({ orderTx: '123' })
+    expect(orders).toStrictEqual(
```

Comment:
this is brittle maybe check that the order list is empty rather than exact object with error message